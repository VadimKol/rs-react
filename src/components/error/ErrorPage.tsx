import { type ReactNode, useEffect, useState } from 'react';
import { isRouteErrorResponse, useRouteError } from 'react-router-dom';

import error_img from '@/assets/images/loader.png';
import { useTheme } from '@/hooks/useTheme';

import styles from './styles.module.scss';

export function ErrorPage(): ReactNode {
  const error = useRouteError();
  const [errorMessage, setErrorMessage] = useState<string>();
  const { theme } = useTheme();

  useEffect(() => {
    if (error) {
      if (isRouteErrorResponse(error)) {
        setErrorMessage(error.statusText);
      } else if (error instanceof Error) {
        setErrorMessage(error.message);
      } else {
        setErrorMessage(String(error));
      }
    }
  }, [error]);

  return (
    <main className={theme === 'dark' ? 'main' : 'main light'}>
      <div className={styles.error}>
        <p className={styles.name}>ERROR</p>
        <img className={styles.error_img} src={error_img} alt="Rick and Morty error" width={500} height={500} />
        <p className={styles.desc}>{errorMessage}</p>
        <p className={styles.desc}>Please refresh the page</p>
      </div>
    </main>
  );
}
