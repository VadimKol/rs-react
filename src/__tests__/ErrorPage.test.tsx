import { render } from '@testing-library/react';

import ErrorPage from '@/pages/_error';

describe('ErrorPage Component', () => {
  it('renders correctly', () => {
    const { container } = render(<ErrorPage error={new Error('Some error')} />);

    expect(container).toMatchSnapshot();
  });
});
