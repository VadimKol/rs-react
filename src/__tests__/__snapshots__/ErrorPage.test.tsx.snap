// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage Component renders correctly 1`] = `
<div>
  <main
    class="main"
  >
    <div
      class="error"
    >
      <p
        class="name"
      >
        ERROR
      </p>
      <img
        alt="Rick and Morty error"
        class="error_img"
        height="500"
        src="stub-test"
        width="500"
      />
      <p
        class="desc"
      />
      <p
        class="desc"
      >
        Please refresh the page
      </p>
    </div>
  </main>
</div>
`;
